lib/zenodo/harvester.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Zenodo do
+  defmodule Harvester do
+    def get_zenodo_records(search_url) do
+      ZenodoApiStream.response_pages(search_url: search_url)
+      |> Enum.flat_map(&parse_zenodo_records/1)
+    end
+
+    def get_zenodo_fields(search_url) do
+      ZenodoApiStream.response_pages(search_url: search_url)
+      |> Enum.reduce([], &accumulate_fields/2)
```

Comment:
I think you can do something like

```elixir
ZenodoApiStream.response_pages(search_url: search_url)
|> Stream.flat_map(&get_hits/1)
|> Enum.flat_map(&extract_fields/1)
```

Reply:
This suggestions works and is more compact.